// postcss/lib/rule.js
'use strict';

var _interopRequireWildcard = function (obj) { return obj && obj.__esModule ? obj : { 'default': obj }; };

var _classCallCheck = function (instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } };

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _inherits = function (subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) subClass.__proto__ = superClass; };

exports.__esModule = true;

var _Container2 = require('./container');

var _Container3 = _interopRequireWildcard(_Container2);

var _list = require('./list');

var _list2 = _interopRequireWildcard(_list);

var Rule = (function (_Container) {
    function Rule(defaults) {
        _classCallCheck(this, Rule);

        _Container.call(this, defaults);
        if (!this.nodes) this.nodes = [];
        this.type = 'rule';
    }

    _inherits(Rule, _Container);

    Rule.prototype.stringify = function stringify(builder) {
        this.stringifyBlock(builder, this.stringifyRaw('selector'));
    };

    _createClass(Rule, [{
        key: 'selectors',
        get: function () {
            return _list2['default'].comma(this.selector);
        },
        set: function (values) {
            this.selector = values.join(', ');
        }
    }]);

    return Rule;
})(_Container3['default']);

exports['default'] = Rule;
module.exports = exports['default'];
